movie_search/media.py

```diff
@@ -78,96 +110,99 @@ def store_data(self, data):
+        movie_recommendations = self.store_recommendations(
+            movie.movie_id,
+        )
+        movie.recommendation.add(*movie_recommendations)
+
+        videos = self.store_videos(movie, videos_data)
 
+        return movie, videos
+    
     def store_recommendations(self, tmdb_id):
         recommendations_data = tmdb_api_obj.get_data_from_endpoint(
             f"/movie/{tmdb_id}/recommendations"
         )
-        movie_recommendations = []
-
+        recommendations = []
         for rec_data in recommendations_data.get("results", []):
             recommendation, created = MovieRecommendation.objects.get_or_create(
-                tmdb_id=rec_data.get("id", 0),
+                movie_id=rec_data.get("id", 0),
                 poster_path=rec_data.get("poster_path", ""),
             )
-            movie_recommendations.append(recommendation)
-        return movie_recommendations
+            recommendations.append(recommendation)
+        return recommendations
+    
+    def store_videos(self, media_obj, videos_data):
+        for video_data in videos_data.get("results", []):
+            MovieVideo.objects.get_or_create(
+                movie=media_obj,
+                name=video_data.get("name", ""),
+                key=video_data.get("key", ""),
+            )
+        return MovieVideo.objects.filter(movie_id=media_obj.id)
+
+    def get_movie_discover_data(self, **kwargs):
+        # Process genres
+        genre_names = kwargs.get("genre_names", [])
+        genres = self.get_genres_from_discover(genre_names, MovieGenre)
+
+        # Process person
+        person_name = kwargs.get("person_name")
+        person_id = None
+        if person_name:
+            person_service = PersonService()
+            person_data = person_service.fetch_person_data(person_name)
+            person_id = person_service.get_person_id(person_data, person_name)
+
+        # Process providers
+        watch_provider_names = kwargs.get("watch_provider_names", [])
+        providers = self.get_providers_from_discover(
+            watch_provider_names, MovieProvider
+        )
 
-    def get_genres_from_discover(self, genre_names):
-        genres = MovieGenre.objects.filter(name__in=genre_names).values_list(
-            "genre_id", flat=True
+        # Update kwargs with processed data
+        kwargs.update(
+            {
+                "with_genres": ",".join(map(str, genres)),
+                "with_watch_providers": ",".join(map(str, providers)),
+                "with_people": person_id,
+            }
         )
-        return genres
 
-    def get_providers_from_discover(self, watch_provider_names):
-        providers = MovieProvider.objects.filter(
-            name__in=watch_provider_names
-        ).values_list("provider_id", flat=True)
-        return providers
+        # Remove None values from kwargs
+        kwargs = {k: v for k, v in kwargs.items() if v is not None}
```

Comment:
can remove comment, also done below, so might deduplicate it in a helper function